src/main/java/com/atlassian/db/replica/internal/LazyReference.java

```diff
@@ -1,31 +1,36 @@
 package com.atlassian.db.replica.internal;
 
 
-import java.util.concurrent.atomic.AtomicReference;
+import com.atlassian.db.replica.internal.util.ThreadSafe;
+
 import java.util.function.Supplier;
 
+@ThreadSafe
 public abstract class LazyReference<T> implements Supplier<T> {
```

Comment:
- [ ]  hide the change with `compatibleWithPreviousVersion`.

Reply:
Yep, let's hide it

Reply:
@wyrzyk I think it's done, right?